### DISCUSSION TITLE
Blazor Server and Application Context

### DISCUSSION THREAD

**Discussion by mikeclimbs23** [MarimerLLC/csla#2595] [Questions]
Hello,

I'm using CSLA 5.5 with blazor server. I store the database name in the applicationContext.GlobalContext.  Csla.ApplicationContext.GlobalContext.Add("Database", DatabaseName);

In bizoject validation, I load a bizobject to check validation meets certain user-definable rules. I'm loading the bizobject using the code below.
CO_CompanyInfo CompanyInfo;
CompanyInfo = Csla.DataPortal.Fetch<FezWeb.BizObj.CO_CompanyInfo>(0)

When the bizobject company info reads the globalcontext the key is missing. I'm guessing I'm not instantiating the biz object correctly and that's why it's referencing an empty ApplicationContext.  Any pointers would be appreciated.

Thanks,

Mike

**Comment by rockfordlhotka** [comment]
Hi @mikeclimbs23 - unfortunately this is a known issue with SSB and CSLA 5. The fix (major rewrite really) is coming in CSLA 6.

The reason the solution is major, is that unlike all .NET frameworks to date, server-side Blazor does something new and interesting: using a dependency injection scope for isolation between user sessions on the server.

In the past, things like the user identity and other state were maintained using thread-local storage or (in aspnet) HttpContext. Those ambient values can be accessed via static code such as `Csla.ApplicationContext.User` or `Csla.ApplicationContext.ClientState`.

In server-side Blazor (not Blazor wasm, which works "like normal"), the only way to get things like the current user identity, or to maintain state, is to use a scope-level service injected via DI.

As it is impossible to inject values into static code, the entire concept of the static methods on ApplicationContext has become unworkable.

So CSLA 6 now has an _instance_ of ApplicationContext that is a service, and it is injected into your code when necessary.

There is (and won't be) a fix in CSLA 5, because these changes are _substantial_ and therefore are breaking to all existing code. We use semantic versioning, so breaking changes require a major version change - hence the "fix" (new feature?) is coming in CSLA 6.

**Reply by mikeclimbs23** [reply]
Thanks Rocky,

I'm kind of happy to hear I wasn't doing something stupid. I'm not in a rush, but is there an eta on CSLA 6?

Thanks,

Mike

**Reply by rockfordlhotka** [reply]
We are closing in on the first prerelease of CSLA 6. The BlazorExample app and a couple other samples are now running, so I'm feeling almost confident enough to put a preview in NuGet.

In terms of an actual 6.0.0 release, that may be another couple months, depending on how rapidly I and other contributors can get everything done.

**Reply by mikeclimbs23** [reply]
Thanks,  I'll keep an eye out for the preview.

Mike
